### ISSUE TITLE
yodaStyleExpr should support in-range operations

### ISSUE THREAD

**Issue by korya** [go-critic/go-critic#863]
Right now `yodaStyleExpr` rule does not allow the following code:

```go
func IsValidHeading(angle float32) bool {
    // The line below is good and should be accepted
    return 0 <= angle && angle <= 360
}

func isInvalidHeading(angle float32) bool {
    // The line below is OK and IMHO should be accepted. But this one is arguable.
    return angle < 0 || 360 > angle
}
```

I think when testing whether a number lies in a given range the code below is good and should be considered a good style.

**Comment by quasilyte** [comment]
I see. Yes, probably makes sense.

**Comment by seiyab** [comment]
Maybe resolved in https://github.com/go-critic/go-critic/pull/1105?
But I'm considering that warning `>=` and `>` can make sense.

**Comment by ma91n** [comment]
```


```sh
$ git diff
diff --git a/checkers/testdata/yodaStyleExpr/negative_tests.go b/checkers/testdata/yodaStyleExpr/negative_tests.go
index 50bede0..737bc0d 100644
--- a/checkers/testdata/yodaStyleExpr/negative_tests.go
+++ b/checkers/testdata/yodaStyleExpr/negative_tests.go
@@ -36,3 +36,13 @@ func _() {
                // character range ok
        }
 }
+
+func IsValidHeading(angle float32) bool {
+       // The line below is good and should be accepted
+       return 0 <= angle && angle <= 360
+}
+
+func isInvalidHeading(angle float32) bool {
+       // The line below is OK and IMHO should be accepted. But this one is arguable.
+       return angle < 0 || 360 > angle
+}

 make test-checker yodaStyleExpr
go test -v -count=1 -run=/yodaStyleExpr ./checkers
=== RUN   TestCheckers
=== RUN   TestCheckers/yodaStyleExpr/debug
=== RUN   TestCheckers/yodaStyleExpr/sanity
=== RUN   TestCheckers/yodaStyleExpr
--- PASS: TestCheckers (0.30s)
    --- PASS: TestCheckers/yodaStyleExpr/debug (0.10s)
    --- PASS: TestCheckers/yodaStyleExpr/sanity (0.10s)
    --- PASS: TestCheckers/yodaStyleExpr (0.10s)
=== RUN   TestIntegration
--- PASS: TestIntegration (0.19s)
=== RUN   TestTags
--- PASS: TestTags (0.00s)
=== RUN   TestDocs
--- PASS: TestDocs (0.00s)
=== RUN   TestStableList
--- PASS: TestStableList (0.00s)
=== RUN   TestExternal
    checkers_test.go:164: temporary disabled during bump to Go 1.20
--- SKIP: TestExternal (0.00s)
PASS
ok      github.com/go-critic/go-critic/checkers 0.791s
```
